objects-8/task2.js

```diff
@@ -0,0 +1,189 @@
+    let greatestId = Math.max(...enterprises.map(company => (company.id + company.departments.length)));
+    let deptId = greatestId + 1;
+    if (company.id === companyId) {
+      company.departments.push(
+        {
+          id: deptId,
+          name: deptName,
+        },
+      )
+    }
+  }
+  return enterprises 
+}
+
+
+// 5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+// Пример:
+// editEnterprise(1, "Новое название предприятия")
+function editEnterprise(companyId, companyEditedName){
+ // 1 
+  for (let company of enterprises){
+    if (company.id == companyId){
+       company.name = companyEditedName
+      }
+  }return enterprises;
+
+ // 2
+//  return enterprises.forEach(company => {
+//   if (company.id === companyId){
+//     company.name = companyEditedName
+//   }
+// })
+}
+
+// 6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+// Пример:
+// editDepartment(7, "Новое название отдела")
+function editDepartment(deptId, deptNewName){
+  for (let entreprice of enterprises){
+    entreprice.departments.forEach(department => {
+    if (department.id == deptId){
+      department.name = deptNewName
+    }
+
+  })
+}return enterprises;
+}
+
+// 7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+function deleteEnterprise(entrepriceId){
+  let indexOfCompany = enterprises.find(company => company.id === entrepriceId);
+  enterprises.splice(indexOfCompany, 1);
+  return enterprises;
+}
+
+// 8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+// // Пример:(deleteDepartment(10))
+function deleteDepartment(departmentId){
+  for(let company of enterprises){
+    let indexOfDepartment = company.departments.findIndex(department => department.id == departmentId);
+    if ( indexOfDepartment != -1 && !company.departments[indexOfDepartment].employees_count > 0){
+    company.departments.splice(indexOfDepartment, 1);
+    console.log(`department with id ${departmentId} is deleted!`)
+ }}
+return enterprises;
+}
+
+// 9. Написать функцию для переноса сотрудников между отделами одного предприятия. В качестве аргумента принимает два значения: 
+//id отдела, из которого будут переноситься сотрудники и id отдела, в который будут переноситься сотрудники).
+// Пример:
+// moveEmployees(2, 3)
+function moveEmployees(fromDeptId, toDeptId){
+  for (let company of enterprises){
+    let indexFrom = company.departments.findIndex(department => department.id === fromDeptId);
+    let indexTo = company.departments.findIndex(department => department.id === toDeptId);
+      if (indexFrom != -1 && indexTo != -1){
+      let howManyPeopleTransfer = company.departments[indexFrom].employees_count;
+      company.departments[indexTo].employees_count +=  howManyPeopleTransfer;
```

Comment:
You forgot to remove people from "indexFrom"

Reply:
removed people)